spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -455,141 +423,148 @@ def add_electrodes(recording: se.RecordingExtractor, nwbfile=None,
-
+        defaults = dict(
+            id=np.nan,
+            x=np.nan,
+            y=np.nan,
+            z=np.nan,
+            # There doesn't seem to be a canonical default for impedence, if missing.
+            # The NwbRecordingExtractor follows the -1.0 convention, other scripts sometimes use np.nan
+            imp=-1.0,
+            location="unknown",
+            filtering="none",
+            group_name="Electrode Group"
+        )
         if metadata is None:
-            metadata = dict()
-
-        metadata_columns = []
-        if len(metadata.keys()) > 0:
-            if 'Ecephys' in metadata and 'Electrodes' in metadata['Ecephys']:
-                if type(metadata['Ecephys']['Electrodes']) is list and metadata['Ecephys']['Electrodes']:
-                    metadata_columns = metadata['Ecephys']['Electrodes']
-                    for j, metadata_column in enumerate(metadata_columns):
-                        if type(metadata_column) is dict \
-                                and set(metadata_column.keys()) == set(['name', 'description', 'data']) \
-                                and type(metadata_column['data']) is list:
-                            if metadata_column['name'] != 'group' and metadata_column['name'] != 'group_name':
-                                nwbfile.add_electrode_column(str(metadata_column['name']),
-                                                             str(metadata_column['description']))
-                        else:
-                            print(f"Warning: Expected metadata['Ecephy']['Electrodes'][{j}] to be"
-                                  " a dictionary with keys 'name', 'description', and 'data',"
-                                  " with 'data' being a list of items!" +
-                                  " The custom column will not be added.")
-                else:
-                    metadata_columns = []
-                    print("Warning: metadata must be a list of dictionaries of the form"
-                          " metadata['Ecephys']['Electrodes'] = [{'name': my_name,"
-                          " 'description': my_description, 'data': [my_electrode_data]}, ...]"
-                          " where [my_electrode_data] is a list in one-to-one correspondence with"
-                          " the nwbfile electrode ids and RecordingExtractor channel ids!")
-
-        for j, channel_id in enumerate(channel_ids):
-            # Will not overwrite the electrode id if already in nwbfile
+            metadata = dict(
+                Ecephys=dict(
+                    Electrodes=[]
+                )
+            )
+        assert all([isinstance(x, dict) and set(x.keys()) == set(['name', 'description', 'data'])
+                    and isinstance(x['data'], list) for x in metadata['Ecephys']['Electrodes']]), \
+            "Expected metadata['Ecephys']['Electrodes'] to be a list of dictionaries!"
```

Comment:
If input validation is getting that complex, it might be better to just use a json schema validator